extensions/widgets/switchbutton/switchbutton.lcb

```diff
@@ -181,35 +379,303 @@ end handler
+		setSwitch(not(mSwitchIsOn))
+		put false into mAnimating
+		redraw all
+	end if
+
+	post "mouseUp" with [mSwitchIsOn]
+end handler
+
+--------------------------------------------------------------------------------
+--
+--		Setting Properties
+--
+--------------------------------------------------------------------------------
+
+constant kWidgetThemeIos is "iOS"
+constant kWidgetThemeAndroidDark is "Android(Dark)"
+constant kWidgetThemeAndroidLight is "Android(Light)"
+
+private handler setWidgetTheme(in pWidgetTheme as String) returns nothing
+
+	variable tThemeList as List
+	variable tInList as Boolean
+
+	put [kWidgetThemeIos, kWidgetThemeAndroidDark, kWidgetThemeAndroidLight] into tThemeList
+	put pWidgetTheme is in tThemeList into tInList
+
+	if tInList is false then
+		throw "invalid widget theme"
+	end if
+
+	if pWidgetTheme is mWidgetTheme then
+		return
+	end if
+
+	put pWidgetTheme into mWidgetTheme
+	if mWidgetTheme contains "Android" then
+		put "Teal" into mColorScheme
+	else
+		put the empty string into mColorScheme
 	end if
+	redraw all
+end handler
+
+constant kAndroidColorSchemeRed is "Red"
+constant kAndroidColorSchemePink is "Pink"
+constant kAndroidColorSchemePurple is "Purple"
+constant kAndroidColorSchemeDeepPurple is "Deep Purple"
+constant kAndroidColorSchemeIndigo is "Indigo"
+constant kAndroidColorSchemeBlue is "Blue"
+constant kAndroidColorSchemeLightBlue is "Light Blue"
+constant kAndroidColorSchemeCyan is "Cyan"
+constant kAndroidColorSchemeTeal is "Teal"
+constant kAndroidColorSchemeGreen is "Green"
+constant kAndroidColorSchemeLightGreen is "Light Green"
+constant kAndroidColorSchemeLime is "Lime"
+constant kAndroidColorSchemeYellow is "Yellow"
+constant kAndroidColorSchemeAmber is "Amber"
+constant kAndroidColorSchemeOrange is "Orange"
+constant kAndroidColorSchemeDeepOrange is "Deep Orange"
+constant kAndroidColorSchemeBrown is "Brown"
+constant kAndroidColorSchemeGrey is "Grey"
+constant kAndroidColorSchemeBlueGrey is "Blue Grey"
+
+private handler setColorScheme(in pColorScheme as String) returns nothing
+
+	variable tColorSchemeList as List
+	variable tInList as Boolean
+
+	put [kAndroidColorSchemeRed, kAndroidColorSchemePink, kAndroidColorSchemePurple, kAndroidColorSchemeDeepPurple, kAndroidColorSchemeIndigo, kAndroidColorSchemeBlue, kAndroidColorSchemeLightBlue, kAndroidColorSchemeCyan, kAndroidColorSchemeTeal, kAndroidColorSchemeGreen, kAndroidColorSchemeLightGreen, kAndroidColorSchemeLime, kAndroidColorSchemeYellow, kAndroidColorSchemeAmber, kAndroidColorSchemeOrange, kAndroidColorSchemeDeepOrange, kAndroidColorSchemeBrown, kAndroidColorSchemeGrey, kAndroidColorSchemeBlueGrey] into tColorSchemeList
+	put pColorScheme is in tColorSchemeList into tInList
+
+	if pColorScheme is mColorScheme then
+		return
+	end if
+
+	if mWidgetTheme contains "Android" then
+		put pColorScheme into mColorScheme
+	else
+		put the empty string into mColorScheme
+	end if
+	redraw all
+
```

Comment:
If you put the
```
if pColorScheme is mColorScheme then
  return
end if
```

at the beginning of this, then it doesn't need to create the list if the color scheme is set to what it already is.
Also you have missed throwing if tInList is false. In fact, why not just do 

```
if not (pColorScheme is in tColorSchemeList) then
  throw ...
end if
```

then you don't need `tIsInList` at all.
(I think most of this applies to the widget theme property too)